lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -169,7 +193,12 @@ def add_payment_method(post, payment_method, options, key = :source)
             post[key][:last_name] = payment_method.last_name if payment_method.last_name
           end
         end
-        unless payment_method.is_a?(String)
+        if payment_method.is_a?(String)
```

Comment:
💬  Comment:

Just to follow the previous logi, shouldn't be rasonable here to use:

```ruby
  elsif payment_method.is_a?(String)
```

or maybe is a good time for a refactor and add a case statement here ?

Reply:
I'll try with case statements.